backend/functions/healthcheck/healthcheck.db.cosmos.ts

```diff
@@ -31,57 +45,61 @@ export default class HealthcheckCosmosDb {
 
   public async checkDbRead() {
     try {
-      const { resources: results } = await this.cosmosDbClient
-        .database(this.databaseName)
-        .container(this.CONTAINER_NAME)
-        .items.readAll()
-        .fetchAll();
-      return results.length > 0;
+      const result = await this.getAdapter<HealthCheckDocument>().find(null);
+
+      const items = [];
+      for await (const doc of result) {
+        items.push(doc);
+      }
+      return items.length > 0;
     } catch (e) {
-      this.applicationContext.logger.error(MODULE_NAME, `${e.name}: ${e.message}`);
+      this.context.logger.error(MODULE_NAME, `${e.name}: ${e.message}`);
     }
     return false;
   }
 
   public async checkDbWrite() {
+    const healthCheckDocument: HealthCheckDocument = {
+      healtchCheckId: 'arbitrary-id',
+      documentType: 'HEALTH_CHECK',
+    };
     try {
-      const { resource } = await this.cosmosDbClient
-        .database(this.databaseName)
-        .container(this.CONTAINER_NAME)
-        .items.create({});
-      this.applicationContext.logger.debug(MODULE_NAME, `New item created ${resource.id}`);
-      return resource.id != undefined;
+      const resource = await this.getAdapter<HealthCheckDocument>().insertOne(healthCheckDocument);
+      this.context.logger.debug(MODULE_NAME, `New item created ${resource}`);
+      return !!resource;
     } catch (e) {
-      this.applicationContext.logger.error(MODULE_NAME, `${e.name}: ${e.message}`);
+      this.context.logger.error(MODULE_NAME, `${e.name}: ${e.message}`);
     }
     return false;
   }
 
   public async checkDbDelete() {
+    const { equals } = QueryBuilder;
     try {
-      const { resources: results } = await this.cosmosDbClient
-        .database(this.databaseName)
-        .container(this.CONTAINER_NAME)
-        .items.readAll()
-        .fetchAll();
+      const result = await this.getAdapter().find(null);
 
-      if (results.length > 0) {
-        for (const resource of results) {
-          this.applicationContext.logger.debug(
-            MODULE_NAME,
-            `Invoking delete on item ${resource.id}`,
-          );
+      const items = [];
+      for await (const doc of result) {
+        items.push(doc);
+      }
+
+      if (items.length > 0) {
+        for (const resource of items) {
+          this.context.logger.debug(MODULE_NAME, `Invoking delete on item ${resource.id}`);
 
-          await this.cosmosDbClient
-            .database(this.databaseName)
-            .container(this.CONTAINER_NAME)
-            .item(resource.id, resource.id)
-            .delete();
+          await this.getAdapter().deleteOne(
+            QueryBuilder.build(
+              equals<HealthCheckDocument['healtchCheckId']>(
+                'healthCheckId',
+                resource.healthCheckId,
+              ),
+            ),
+          );
         }
       }
       return true;
```

Comment:
I realize this isn't related to changes we're making here, but I'm not sure we should return true if no items were found to delete. We're asserting that we have the ability to delete, but technically don't know that we do.

Reply:
Also, I don't know that we need the logging.

Reply:
I agree, and was thinking about this yesterday before walking away. I think validating the result with result.deletedCount > 0 would be best here